contracts/test/Gateway.t.sol

```diff
@@ -909,4 +919,53 @@ contract GatewayTest is Test {
         vm.expectRevert(Assets.InvalidDestinationFee.selector);
         IGateway(address(gateway)).sendToken{value: fee}(address(token), destPara, recipientAddress32, 0, 1);
     }
+
+    function testAgentRegisterDot() public {
+        RegisterForeignTokenParams memory params = RegisterForeignTokenParams({
+            agentID: assetHubAgentID,
+            tokenID: dotTokenID,
+            name: "DOT",
+            symbol: "DOT",
+            decimals: 10
+        });
+
+        vm.expectEmit(true, true, false, false);
+        emit IGateway.ForeignTokenRegistered(bytes32(uint256(1)), assetHubAgentID, address(0));
+
+        GatewayMock(address(gateway)).registerForeignTokenPublic(abi.encode(params));
+    }
```

Comment:
Some negative tests would be good to add.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/6148c038b97c2a26982dd0825576438d73a21b44